### PR TITLE
MAISTRA-1831 use objectSelector in injector webhook

### PR DESCRIPTION
**Pull request by rcernich**
Signed-off-by: rcernich <rcernich@redhat.com>

### CODE REVIEW

**Comment by rcernich** [comment]
I haven't tested this yet.  I left the logic for `istio.io/rev` always enabled to allow users to control which injector they want to use for the app.  This allows fine grained control over the upgrade process.  That said, I think we need to put a little more thought into how we will manage revisions going forward.  (I could see using `ServiceMeshMember` to select which control plane (revision) for the namespace and then put the correct `istio.io/rev` label on the namespace as part of SMMR reconciliation.)

**Review by dgn — APPROVED** [review]
Thanks, I think this makes sense.

Regarding the revision name, I think in the future we'll want to make it <smcp.name>-<smcp.generation> so we can actually perform upgrades of the same SMCP using revisions. wdyt?

**Comment by rcernich** [comment]
I was thinking over night that requiring the revision might be a little burdensome for applications, as they'd need to know the name of the SMCP resource they want to use.  Furthermore, this is a break from 1.1, where the only requirement was adding the inject label to their pod.  Given that, I think it's best to remove the revision requirement, possibly adding it to the namespace selector in the future.

Regarding the name-generation, I'm not sure we'd want to do that as I think this could cause pods that were injected to not be injected after an update occurs, because it's referencing the older generation, e.g. if the pod were rescheduled because of cluster maintenance, the new pod would not get injected.

File: resources/smcp-templates/v2.0/base [hunk 1]
```diff
@@ -24,7 +24,7 @@ spec:
     level: warn
 
   proxy:
-    autoInject: true
+    autoInject: disabled
```
Comment (dgn) [inline resources/smcp-templates/v2.0/base]: why `disabled`? I think this has to be `false`, otherwise it will cause the `if` condition to be `true`

Comment (rcernich) [reply resources/smcp-templates/v2.0/base]: doh! mixing contexts: new smcp and old values.yaml.  that saved an extra build/test cycle.  thanks!

**Review by dgn — APPROVED** [review]
just needs a rebase now

**Comment by rcernich** [comment]
And I need to test it too. ;)

**Pull request merged**